### ISSUE TITLE
Memory-Leak  in AcquireSemaphoreMemory MagickCore/semaphore.c:154

### ISSUE THREAD

**Issue by zhouat** [ImageMagick/ImageMagick#634]
Poc link :
https://github.com/zhouat/poc_IM/blob/master/Memory-Leak-21_output_picon_1501391824.23

Trigger Command：` magick convert Memory-Leak-21_output_picon_1501391824.23 out.picon `

```
...
...

Indirect leak of 64 byte(s) in 1 object(s) allocated from:
    #0 0x7f89eef2d076 in __interceptor_posix_memalign (/usr/lib/x86_64-linux-gnu/libasan.so.2+0x99076)
    #1 0x7f89ee751aa5 in AcquireSemaphoreMemory MagickCore/semaphore.c:154
    #2 0x7f89ee751b61 in AcquireSemaphoreInfo MagickCore/semaphore.c:200
    #3 0x7f89ee4a845a in AcquirePixelCache MagickCore/cache.c:226
    #4 0x7f89ee6324f6 in AcquireImage MagickCore/image.c:206
    #5 0x7f89ee84146a in ReadGIFImage coders/gif.c:996
    #6 0x7f89ee516ed4 in ReadImage MagickCore/constitute.c:497
    #7 0x7f89ee49118b in BlobToImage MagickCore/blob.c:428
    #8 0x7f89ee97cf40 in WritePICONImage coders/xpm.c:689
    #9 0x7f89ee51a032 in WriteImage MagickCore/constitute.c:1114
    #10 0x7f89ee51acae in WriteImages MagickCore/constitute.c:1333
    #11 0x7f89ede735cf in ConvertImageCommand MagickWand/convert.c:3280
    #12 0x7f89edef370b in MagickCommandGenesis MagickWand/mogrify.c:183
    #13 0x4017e1 in MagickMain utilities/magick.c:149
    #14 0x4019c2 in main utilities/magick.c:180
    #15 0x7f89ed71f82f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2082f)

```

**Comment by mikayla-grace** [comment]
Unfortunately, we cannot reproduce this problem. Its possible the fix for #631 also fixed this problem as well.

**Comment by bastien-roucaries** [comment]
CVE-2017-11755
